packages/node_modules/@node-red/editor-client/src/js/settings.js

```diff
@@ -228,12 +228,70 @@ RED.settings = (function () {
             return defaultValue;
         }
     }
-    function getLocal(key) {
-        return localStorage.getItem(key)
+    /**
+     * Retrieve a JSON-deserialized value from localStorage for the given key.
+     *
+     * The stored value is expected to be a JSON-serialised representation (for
+     * example produced by JSON.stringify).
+     *
+     * @param {string} key The localStorage key to read.
+     * @param {unknown} [defaultValue] Value returned when the key is missing,
+     * parsing fails, or the parsed value is `undefined`. Optional.
+     * @returns {unknown|undefined} The parsed value from localStorage,
+     * or `defaultValue` if not available or on parse failure.
+     *
+     * @example
+     * // If localStorage.getItem('settings') === '{"theme":"dark"}'
+     * const settings = RED.settings.getLocal('settings', {});
+     * // settings -> { theme: 'dark' }
+     *
+     * @example
+     * // If key is missing or contains invalid JSON
+     * const count = RED.settings.getLocal('visitCount', 0);
+     * // count -> 0
+     */
+    function getLocal(key, defaultValue) {
+        try {
+            const value = JSON.parse(localStorage.getItem(key) ?? "undefined");
+            if (value !== undefined) {
+                return value;
+            }
+            return defaultValue;
+        } catch (error) {
+            console.warn("settings.getLocal: parse error: ", error);
+            return defaultValue;
+        }
     }
```

Comment:
```js
function getLocal(key, defaultValue) {
    const storedValue = localStorage.getItem(key);
    if (stored === null) {
        // NOTE: when not set local storage returns null
        return defaultValue;
    }

   // NOTE: parse only when really needed
    try {
        return JSON.parse(stored);
    } catch (error) {
        // NOTE: capture exception only when really needed. You should not rely on exception for flow control because it is expensive
        console.warn("settings.getLocal: parse error:", error);
        return defaultValue;
    }
}
```